### ISSUE TITLE
Select 2 allowClear option not working

### ISSUE THREAD

**Issue by sharky98** [tetranz/select2entity-bundle#10]
Hi,
I've been trying to get the Select2 allowClear option to be active. However, I didn't seem to find the way to activate it with your select2entity-bundle. I've tried  to add `$("#myField").select2entity({allowClear: true});` or `$("#myField").select2({allowClear: true});`, both before and after I've included the `select2entity.js` file. Either way, it's not working. It return the error `Uncaught TypeError: Cannot read property 'slice' of undefined` in `select2.min.js`.

May I suggest that Select2Entity act similar to Select2, in the way that it doesn't auto-activate the Select2 form (at line 25 of /Resources/public/js/select2entity.js). Instead, let the user activate similar to Select2, with something like `$("#myField").select2entity();`, passing all the attributes to the `.select2();`. Moreover, this auto-activatation, prevent applying the Select2 defaults to Select2Entity (since the default would most probably set after including the select2entity.js file).

**Comment by sharky98** [comment]
Here a version of `select2entity.js` that seems to do the trick. The user must now activate the field (which he have to do anyway with any other select2 field), with something like `$("#myField").select2entity();`
```javascript
$(document).ready(function () {
  $.fn.select2entity = function (action) {
    var select2entityParam = {
      ajax: {
        data: function (params) {
          return {
            q: params.term
          };
        },
        processResults: function (data) {
          return {
            results: data
          };
        },
        cache: true
      }
    };
    $.extend(select2entityParam, action);
    this.select2(select2entityParam);
    return this;
  };
});
```

**Comment by tetranz** [comment]
Thanks. I want to take a closer look at this when I have more time. Maybe this weekend.

Perhaps someone else might have some comments.

**Comment by tetranz** [comment]
Try dev-master now. I did some updates today that include allow_clear as an option.  I didn't change the JS. I added it as a data-allow-clear attribute like the other options. I think that's easier. allow_clear simply got missed earlier.

While I was at it, I added a couple more options, delay and cache, and updated the documentation for Symfony 3.

**Comment by bnd170** [comment]
It works perfectly. This bundle is exactly i needed :) thanks so much

**Comment by tetranz** [comment]
Thanks. I've just released version 2.3 with this included.

**Comment by sharky98** [comment]
Thanks, this is working. However I needed also tag support. I'll continue the discussion in #13.
